app/main.py

```diff
@@ -1,3 +1,14 @@
 def cache(func):
-    # Write your code here
-    pass
+    functions = dict()
+    arguments = set()
+
+    def inner(*args):
+        if func.__name__ in functions and args in arguments:
```

Comment:
You don't actually need to handle function names, because there are separate dict/set for each decorated function created in lines 2-3.